app/controllers/api/v1x0/requests_controller.rb

```diff
@@ -22,32 +22,37 @@ def show
       end
 
       def index
-        requests = if params[:request_id]
-                     resource_check('read', params[:request_id], Request) # NotAuthorizedError if current user cannot access parent request
-                     Request.find(params[:request_id]).children
-                   else
-                     Request.where(:parent_id => nil)
-                   end
-
-        collection(index_scope(requests))
+        collection(index_scope(requests_prefilter))
       end
 
       private
 
+      def requests_prefilter
+        return nil unless params[:request_id]
+
+        resource_check('read', params[:request_id], Request) # NotAuthorizedError if current user cannot access parent request
+        Request.find(params[:request_id]).children
+      end
+
       def rbac_scope(relation)
-        ids =
-          case Insights::API::Common::Request.current.headers[Insights::API::Common::Request::PERSONA_KEY]
-          when PERSONA_ADMIN
-            raise Exceptions::NotAuthorizedError, "No permission to access the complete list of requests" unless admin?
-          when PERSONA_APPROVER
-            raise Exceptions::NotAuthorizedError, "No permission to access requests assigned to approvers" unless approver?
-
-            approver_id_list(relation.model.table_name)
-          when PERSONA_REQUESTER, nil
-            owner_id_list(relation.model.table_name)
-          else
-            raise Exceptions::NotAuthorizedError, "Unknown persona"
-          end
+        ids = nil
+        case Insights::API::Common::Request.current.headers[Insights::API::Common::Request::PERSONA_KEY]
```

Comment:
@bzwei You had mentioned that you are going to be removing the personas, so do we still want to implement a solution with the personas?

Reply:
Yes. We are going to remove personas once capabilities are introduced. For now we still need to maintain the code as is.